sw/device/lib/dif/dif_aes.h

```diff
@@ -0,0 +1,589 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_AES_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_AES_H_
+
+/**
+ * @file
+ * @brief <a href="/hw/ip/aes/doc/">AES</a> Device Interface Functions
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_warn_unused_result.h"
+
+// Header Extern Guard (so header can be used from C and C++)
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * This API assumes transactional nature of work, where the peripheral is
+ * configured once per message (data consisting of 1..N 128-bit blocks), and
+ * then "de-initialised" when this message has been fully encrypted/decrypted.
+ *
+ * The peripheral is configured through one of the cipher mode "start"
+ * functions:
+ * `dif_aes_start_ecb`, `dif_aes_start_cbc`, ... .
+ *
+ * Then the encryption/decryption data is fed one 128-bit block at the
+ * time through `dif_aes_load_data` function. The cipher mode operation details
+ * are described in the description of above mentioned "start" functions. When
+ * configured in "automatic" operation mode, every "load data" call, will
+ * trigger encryption/decryption. This is not true when in "manual" operation
+ * mode, where encryption/decryption is triggered by explicitly setting the
+ * `aes.TRIGGER.START` flag through `dif_aes_trigger` call.
+ *
+ * When an entire requested message has been processed the internal state of
+ * AES registers must be securely cleared, by calling `dif_aes_end`.
+ *
+ * Please see the following documentation for further information:
+ * https://docs.opentitan.org/hw/ip/aes/doc/
+ * https://csrc.nist.gov/csrc/media/publications/fips/197/final/documents/fips-197.pdf
+ * https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-38a.pdf
+ */
+
+/**
+ * A typed representation of the AES key share.
+ *
+ * Two part masked AES key, where XOR operation of these two parts results in
+ * the actual key.
+ */
+typedef struct dif_aes_key_share {
+  /**
+   * One share of the key that when XORed with `part1` results in the actual
+   * key.
+   */
+  uint32_t part0[8];
```

Comment:
I think you wanted to change this to `share0` similarly `share1` below. Did you maybe forget to commit/push?

Reply:
Possibly, thanks.

Done.